SparkPost/2web2ui
# TR-1358 - Templates list page

Implements templates list view based on mock. As discussed, using `templatesV2` directory and `templatesv2` new route to namespace all v2 work and minimize cleanup/diff for release. This PR is based on #1032; merging that first would reduce diff. 

### What Changed
 - Created a new directory `templatesV2` to keep new designs. 
- In order to minimize cleanup later, refactored routes to support both templates and templatesV2 routing. Any routes missing in v2, will fallback to v1 (/templates). 

### How To Test
 - Checkout to branch
 - Visit `Templates` from `Campaigns` menu. It should take you to new list page. Rest should work as is. 
 - Note: If you click breadcrumbs (like from template edit page), it'll take to old templates route. Those can be fixed after `next` editing paths and breadcrumb links are updated in editing PRs. 


### To Do
- [x] Displaying separate rows for draft & publish
- [x] Implement delete
- [x] Update description 

- [x] Change base branch to `master` before merging

Notes:
- Brian used `/next` suffix for all of his routing. As we discussed this, in a later PR, i'll modify that and bring under `/templatev2`
